### PR TITLE
Extract inline js

### PR DESCRIPTION
**Pull request by SuperTux88**
I extracted some of the inline JS-code to external JS-files.
- Tags autocomplete: I unified the code for the tags autocompletion, so it isn't copied everywhere, because it was mainly the same.
- Profile photo uploader: Just copied to a JS file and use translations from JS-I18n, so I could delete the other (duplicate) translations.
- Trigger the getting started popovers from backbone via gon.
- Trigger background search timer with gon
- Extract mobile bookmarklet JS to separate file and use the already available data from gon.

I also removed some unused JS-files from the precompile list.

### CODE REVIEW

File: app/assets/javascripts/mobile/mobile_profile_edit.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$(document).ready(function() {
```
Comment (diaspora-code-review) [inline app/assets/javascripts/mobile/mobile_profile_edit.js]: Block must not be padded by blank lines.

**Comment by denschub** [comment]
@svbergerem, could you have a look please?

**Comment by denschub** [comment]
(sorry, hit the wrong button. didn't mean to close.)

File: app/assets/javascripts/helpers/tags_autocomplete.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      selectedValuesProp: "name",
+      searchObjProps: "name",
+      asHtmlID: "tags",
+      neverSubmit: true,
+      retrieveLimit: 10,
+      selectionLimit: false,
+      minChars: 2,
+      keyDelay: 200,
+      startText: "",
+      emptyText: "no_results"
```
Comment (svbergerem) [inline app/assets/javascripts/helpers/tags_autocomplete.js]: This should be `Diaspora.I18n.t("no_results")`.

File: app/assets/javascripts/pages/users-getting-started.js [hunk 3]
```diff
@@ -49,36 +49,13 @@ Diaspora.Pages.UsersGettingStarted = function() {
...
-      minChars: 2,
-      keyDelay: 200,
-      startText: "",
-      emptyText: "no_results",
+    new Diaspora.TagsAutocomplete("#follow_tags", {
       selectionAdded: function(elem){tagFollowings.create({"name":$(elem[0]).text().substring(2)})},
       selectionRemoved: function(elem){
         tagFollowings.where({"name":$(elem[0]).text().substring(2)})[0].destroy();
         elem.remove();
       }
```
Comment (svbergerem) [inline app/assets/javascripts/pages/users-getting-started.js]: Add `preFill: gon.preloads.tagsArray` and add the gon preload to the getting_started method of the users controller.

File: app/views/users/getting_started.mobile.haml [hunk 4]
```diff
@@ -5,31 +5,7 @@
...
-
-      autocompleteInput.bind('keydown', function(evt){
-        if(evt.which === Keycodes.ENTER || evt.which === Keycodes.TAB || evt.which === Keycodes.SPACE) {
-          evt.preventDefault();
-          if( $('li.as-result-item.active').length == 0 ){
-            $('li.as-result-item').first().click();
-          }
-        }
-      });
+      new Diaspora.TagsAutocomplete("#follow_tags");
```
Comment (svbergerem) [inline app/views/users/getting_started.mobile.haml]: Add `preFill: gon.preloads.tagsArray`.

**Review by svbergerem — CHANGES_REQUESTED** [review]

**Review by svbergerem — APPROVED** [review]

**Comment by svbergerem** [comment]
Thank you! :+1:

**Pull request closed without merging**